### PR TITLE
MULE-18232: ObjectStore store operation shouldn't throw ObjectAlreadyExistsException if failIfPresent is set to false

### PR DESCRIPTION
**Pull request by eze210**

### CODE REVIEW

File: src/test/java/org/mule/extension/objectstore/internal/ObjectStoreOperationsTestCase.java [hunk 1]
```diff
@@ -59,4 +71,50 @@ public void retrieveValueWhenObjectStoreAndDefaultValueAreNull() throws ObjectSt
     objectStoreOperations.retrieve("key", null, null);
   }
 
+  @Test
+  public void concurrentStoreOperationsDontLeadToExceptionsWhenFailIsPresentIsSetToFalse() throws InterruptedException {
+    // Different locks to emulate the behavior on CloudHub cluster
```
Comment (fsgonz) [inline src/test/java/org/mule/extension/objectstore/internal/ObjectStoreOperationsTestCase.java]: on cloudhub with more than one worker.

File: src/test/java/org/mule/extension/objectstore/internal/ObjectStoreOperationsTestCase.java [hunk 2]
```diff
@@ -59,4 +71,50 @@ public void retrieveValueWhenObjectStoreAndDefaultValueAreNull() throws ObjectSt
...
+    when(lockFactory.createLock(anyString())).thenAnswer(invocationOnMock -> new ReentrantLock());
+
+    ObjectStore objectStore = new InMemoryObjectStore();
+
+    int numberOfConcurrentStores = 150;
```
Comment (fsgonz) [inline src/test/java/org/mule/extension/objectstore/internal/ObjectStoreOperationsTestCase.java]: set this as a constant field and remove it from the method.

File: src/test/java/org/mule/extension/objectstore/internal/ObjectStoreOperationsTestCase.java [hunk 3]
```diff
@@ -59,4 +71,50 @@ public void retrieveValueWhenObjectStoreAndDefaultValueAreNull() throws ObjectSt
...
+    startProcessingSemaphore.release(numberOfConcurrentStores);
+
+    // Wait for threads completion.
+    for (Thread thread : threadList) {
+      thread.join();
+    }
+
+    assertThat(numberOfExceptions.get(), is(0));
+
+    assert true;
```
Comment (fsgonz) [inline src/test/java/org/mule/extension/objectstore/internal/ObjectStoreOperationsTestCase.java]: why is this needed?

**Comment by eze210** [comment]
--validate
